homeassistant/helpers/entity.py

```diff
@@ -253,8 +254,35 @@ def async_update_ha_state(self, force_refresh=False):
                             'https://goo.gl/Nvioub', self.entity_id,
                             end - start)
 
+        def get_overrides(entity_id: str) -> Dict:
+            """Return a dictionary of overrides related to entity_id.
+
+            Whole-domain overrides are of lowest priorities,
+            then regexp on entity ID, and finally exact entity_id
+            matches are of highest priority.
+            """
+            domain_result = regexp_result = exact_result = {}
+            domain = split_entity_id(entity_id)[0]
+            for key, value in _OVERWRITE.items():
+                key_list = [
+                    single_key.strip() for single_key in key.split(',')]
+                if domain in key_list:
+                    domain_result = value
+                if entity_id in key_list:
+                    exact_result = value
+                else:
+                    for pattern in key_list:
+                        if re.fullmatch(pattern, entity_id):
```

Comment:
What do you think is best here? Regular expression matching or glob style matching? In my opinion, glob would be more clear, but regexp would be more descriptive. Regexp might be more descriptive than what is necessary though. I think this matters especially for the period which already has a significant meaning in entity ids. Using regular expression matching gives the period another significant meaning in this context. If you use glob matching, you only add meaning to the asterisk and the question mark which can't exist in regular entity ids.

Just a though.